src/main/java/io/pravega/perf/WriterWorker.java

```diff
@@ -71,57 +99,105 @@ private String createPayload(int size) {
      */
     public abstract void flush();
 
+    /**
+     * Flush the producer data.
+     */
+    public abstract void close();
+
+
     @Override
     public Void call() throws InterruptedException, ExecutionException, IOException {
         perf.benchmark();
         return null;
     }
 
+
     private class EventsWriter implements Performance {
 
         public void benchmark() throws InterruptedException, IOException {
-            final EventsController eCnt = new EventsController(System.currentTimeMillis(), eventsPerSec);
             for (int i = 0; i < events; i++) {
                 recordWrite(payload, stats::recordTime);
-                eCnt.control(i);
             }
             flush();
         }
     }
 
-    private class EventsWriterRW implements Performance {
+
+    private class EventsWriterSleep implements Performance {
 
         public void benchmark() throws InterruptedException, IOException {
-            final long time = System.currentTimeMillis();
-            final String timeHeader = String.format(TIME_HEADER_FORMAT, time);
-            final EventsController eCnt = new EventsController(time, eventsPerSec);
-            final StringBuilder buffer = new StringBuilder(timeHeader + ", " + workerID + ", " + payload);
-            buffer.setLength(messageSize);
-            for (int i = 0; i < events; i++) {
-                final String header = String.format(TIME_HEADER_FORMAT, System.currentTimeMillis());
-                final String data = buffer.replace(0, TIME_HEADER_SIZE, header).toString();
-                writeData(data);
-                eCnt.control(i);
+            final EventsController eCnt = new EventsController(System.currentTimeMillis(), eventsPerSec);
+            int cnt = 0;
+            while (cnt < events) {
+                int loopMax = Math.min(flushEvents, events - cnt);
+                for (int i = 0; i < loopMax; i++) {
+                    eCnt.control(cnt++, recordWrite(payload, stats::recordTime));
+                }
+                flush();
             }
-            flush();
         }
     }
 
+
     private class EventsWriterTime implements Performance {
 
+        public void benchmark() throws InterruptedException, IOException {
```

Comment:
Here and any other methods that override an interface should have an @Override annotation. 
This avoids the very confusing bug that can occur if the interface changes the method signature, and forgets to update all the impls. If there is a default implementation that will get selected instead of the actual impl and because the code runs without error in can be confusing why something that was supposed to occur didn't.

Reply:
i think its not required now because there are no classes in which benchmark is overridden; instead , i returning returning the function itself as interface

Reply:
EventsWriterTime.benchmark() overrides Performance.benchmark()

Reply:
now, ts changed as follows :
  private void EventsWriterTime() throws InterruptedException, IOException {
    ....
}

its no more class, its another function assigned as interface Performance.